src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfigBindingGenDriver.cs

```diff
@@ -0,0 +1,106 @@
+{
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/52062", TestPlatforms.Browser)]
+    public partial class ConfigurationBindingGeneratorTests : ConfigurationBinderTestsBase
+    {
+        internal sealed class ConfigBindingGenDriver : IDisposable
+        {
+            private readonly CSharpParseOptions _parseOptions;
+            private readonly CSharpGeneratorDriver _csharpGenerationDriver;
+
+            private readonly AdhocWorkspace _workspace;
+            private Project _project;
+
+            public ConfigBindingGenDriver(
+                LanguageVersion langVersion = LanguageVersion.LatestMajor,
+                IEnumerable<Assembly>? assemblyReferences = null)
+            {
+                assemblyReferences ??= s_compilationAssemblyRefs;
+
+                _parseOptions = new CSharpParseOptions(langVersion)
+                    .WithFeatures(new[] { new KeyValuePair<string, string>("InterceptorsPreview", "") });
+
+                _workspace = RoslynTestUtils.CreateTestWorkspace();
+
+                _project = RoslynTestUtils.CreateTestProject(_workspace, assemblyReferences, langVersion: langVersion)
+                    .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary).WithNullableContextOptions(NullableContextOptions.Annotations))
+                    .WithParseOptions(_parseOptions);
+
+
+                _csharpGenerationDriver = CSharpGeneratorDriver.Create(new[] { new ConfigurationBindingGenerator().AsSourceGenerator() }, parseOptions: _parseOptions);
+            }
+
+            public async Task<ConfigBindingGenResult> RunGeneratorAndUpdateCompilation(params string[]? sources)
+            {
+                if (sources is not null)
+                {
+                    _project = _project.WithDocuments(sources);
+                    Assert.True(_project.Solution.Workspace.TryApplyChanges(_project.Solution));
+                }
+
+                Compilation compilation = (await _project.GetCompilationAsync(CancellationToken.None).ConfigureAwait(false))!;
+                GeneratorDriver generatorDriver = _csharpGenerationDriver.RunGeneratorsAndUpdateCompilation(compilation, out Compilation outputCompilation, out _, CancellationToken.None);
+                GeneratorDriverRunResult runDriverRunResult = generatorDriver.GetRunResult();
+
+                return new ConfigBindingGenResult
+                {
+                    OutputCompilation = outputCompilation,
+                    Diagnostics = runDriverRunResult.Diagnostics,
+                    GeneratedSources = runDriverRunResult.Results[0].GeneratedSources,
+                };
+            }
+
+            public void Dispose() => _workspace.Dispose();
+        }
+
+        internal struct ConfigBindingGenResult
+        {
+            public required Compilation OutputCompilation { get; init; }
+
+            public required ImmutableArray<GeneratedSourceResult> GeneratedSources { get; init; }
```

Comment:
Nit: should this be `GeneratedModel` instead? Judging by the name I expected this to contain source text.